src/main/scala/temple/builder/ServerBuilder.scala

```diff
@@ -27,34 +27,34 @@ object ServerBuilder {
     detail: LanguageDetail,
     projectUsesAuth: Boolean,
   ): ServiceRoot = {
-    val language = serviceBlock.lookupMetadata[ServiceLanguage].getOrElse(ProjectConfig.defaultLanguage)
-    val database = serviceBlock.lookupMetadata[Database].getOrElse(ProjectConfig.defaultDatabase)
+    val language = serviceBlock.lookupMetadata[ServiceLanguage] getOrElse ProjectConfig.defaultLanguage
+    val database = serviceBlock.lookupMetadata[Database] getOrElse ProjectConfig.defaultDatabase
+    val readable =
+      serviceBlock.lookupLocalMetadata[Metadata.Readable] getOrElse ProjectConfig.getDefaultReadable(projectUsesAuth)
+    val writable =
+      serviceBlock.lookupLocalMetadata[Metadata.Writable] getOrElse ProjectConfig.getDefaultWritable(projectUsesAuth)
+
     val languageConfig = language match {
       case ServiceLanguage.Go => GoLanguageConfig
     }
 
     // Whether or not this service has an auth block
     val hasAuthBlock = serviceBlock.lookupLocalMetadata[ServiceAuth].isDefined
 
-    // The created by attribute exists if the project has auth and this service does not have an auth block
-    val createdBy =
-      if (projectUsesAuth && !hasAuthBlock)
```

Comment:
Do we not need this part of the condition?

Reply:
@jaylees14 this is calculated when we add a CreatedByAttribute, so it’s already been done (and the condition is slightly more complex now: struct blocks don’t have createdBy)